share/wake/lib/core/string.wake

```diff
@@ -56,6 +56,27 @@ export def explode (string: String): List String =
     def p s = prim "explode"
     p string
 
+# Extract a substring given a starting and ending codepoint position.
+# The `i`-th codepoint is included if and only if `start <= i < end`.
+# For the purposes of this function, a codepoint is defined as one
+# start byte followed by one or more non-start bytes, where a start
+# byte is any byte with high bits not equal to `10`.
+#
+# Warning: use of this function can divide a multi-codepoint character.
+#
+# Examples:
+# ```
+#   substr "ABCDE" 0 1    = "A"
+#   substr "ABCDE" 1 4    = "BCD"
+#   substr "ABCDE" 1 30   = "BCDE"
+#   substr "ABCDE" 50 3   = ""
+#   substr "ABCDE" (-1) 3 = "ABC"
+#   substr "ABCDE" 1 0    = ""
+# ```
+export def substr (string: String) (start: Integer) (end: Integer): String =
```

Comment:
Actually, do we want the string to be the last argument? That might make it more convenient to use in a String-processing pipeline.

Reply:
Done.